sklearn/decomposition/sparse_pca.py

```diff
@@ -185,6 +191,63 @@ def transform(self, X, ridge_alpha='deprecated'):
         U /= s
         return U
 
+    def get_explained_variance(self, components, X):
+        '''
+            Get the explained variance from the principal components of the
+            data.
+
+            Parameters
+            ----------
+            components : 2D np.ndarray
+                the principal components of shape p x m
+                where m is the number of samples and p the number of
+                components
+            X : 2D np.ndarray
+                The feature vector. Must be shape m x n where m is the number
+                of samples and n is the dimension of the incoming feature
+                vector
+
+            Notes
+            -----
+            The variance ratio may not be computed. The main reason is that we
+            do not know what the total variance is since we did not compute all
+            the components.
+        '''
+        # requires the mean to be subtracted
+        X = X - np.mean(X, axis=0)
+        # the number of samples
```

Comment:
this might not be necessary after #11585 is merged.

Reply:
Sounds good. I'll leave as is for now and rebase on that PR when merged (looks more important than this one right?)

Reply:
It will probably be merged berfore. You can adapt you PR when it's done. Sounds good.